### PR TITLE
Build for both Python 2 and Python 3

### PR DESCRIPTION
**Pull request by phdru**
Build pykcs11_wrap.cpp-py[23] and LowLevel.py[23] at the same time; use
one of the pair (2 or 3) to build PyKCS11 with the current python.

### CODE REVIEW

**Comment by phdru** [comment]
This is work in progress. I'm gonna work on sdist: `cp` commands from 'build-stamp' targets must be implemented in Python in setup.py so that everyone who's running `pip install PyKCS11` got proper versions of the files.

**Comment by phdru** [comment]
Now I think I done what I wanted to achieve.

File: Makefile [hunk 1]
```diff
@@ -37,10 +37,10 @@ src/pykcs11_wrap.cpp-py3: src/pykcs11.i
 src/pykcs11.i: src/opensc/pkcs11.h src/pkcs11lib.h src/pykcs11string.h src/ck_attribute_smart.h
 	touch $@
 
-dist: clean
+dist: src/pykcs11_wrap.cpp-py2 src/pykcs11_wrap.cpp-py3 clean
```
Comment (LudovicRousseau) [inline Makefile]: Be careful. make will generate the 3 targets in parallel if `-j` is used.
Maybe what you want, in fact, is:
```
dist: src/pykcs11_wrap.cpp-py2 src/pykcs11_wrap.cpp-py3
    $(MAKE) clean
```

Comment (phdru) [inline Makefile]: Yep, you're right! And even worse, two parallel SWIG runs stuck with an error — they are trying to overwrite each other's LowLevel.py. I'm working on that…

**Comment by phdru** [comment]
Fixed, `make -j4 dist` works for me now.

**Comment by LudovicRousseau** [comment]
I just noticed that my SWIG Version 3.0.12 does *not* have a `-py3` command line argument (at least according to `swig -help`)

It looks like the same files generated by swig can be used with Python 2 and Python 3. So no need to generate and include different versions of `pykcs11_wrap.cpp` and `LowLevel.py`.

@phdru what are the differences between the `pykcs11_wrap.cpp-py?` and `LowLevel.py?` your generates?

On my side `pykcs11_wrap.cpp-py2` and `pykcs11_wrap.cpp-py3` are identical.
But `PyKCS11/LowLevel.py2` and `PyKCS11/LowLevel.py3` differ a bit. The Python 3 version has just annotation added. i.e.
```
--- PyKCS11/LowLevel.py2    2017-02-15 10:38:41.000000000 +0100
+++ PyKCS11/LowLevel.py3    2017-02-15 10:38:41.000000000 +0100
@@ -96,11 +96,11 @@
     _newclass = 0
 
 
-def cdata(ptr, nelements=1):
+def cdata(ptr: 'void *', nelements: 'size_t'=1) -> "SWIGCDATA":
     return _LowLevel.cdata(ptr, nelements)
 cdata = _LowLevel.cdata
 
-def memmove(data, indata):
+def memmove(data: 'void *', indata: 'void const *') -> "void":
     return _LowLevel.memmove(data, indata)
 memmove = _LowLevel.memmove
 class SwigPyIterator(_object):
```

My proposal: include files generated for Python 2 only and do not use the proposed code to use Python 2 and Python 3.

**Comment by phdru** [comment]
The same for me:

```sh
$ diff pykcs11_wrap.cpp-py[23]

$ diff LowLevel.py[23]
99c99
< def cdata(ptr, nelements=1):
---
> def cdata(ptr: 'void *', nelements: 'size_t'=1) -> "SWIGCDATA":
```

Ok, let's drop this PR. Then you can remove `-py3` from Makefile.

**Pull request closed without merging**